modules/ort/src/main/java/org/eclipse/sw360/antenna/ort/workflow/analyzers/OrtResultAnalyzer.java

```diff
@@ -52,9 +48,19 @@ public String getName() {
         return "OrtResult";
     }
 
-    Collection<Artifact> createArtifactList(InputStream is) throws IOException {
-        ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
-        JsonNode yamlContent = mapper.readTree(is);
+    Collection<Artifact> createArtifactList(File ortResultFile) throws IOException {
+        Optional<String> extension = Optional.ofNullable(ortResultFile.getName())
+                .filter(f -> f.contains("."))
+                .map(f -> f.substring(f.lastIndexOf(".") + 1));
+
+        ObjectMapper mapper;
+        switch (extension.get()) {
+            case "json": mapper = new ObjectMapper(); break;
+            case "yml": mapper = new YAMLMapper(); break;
+            default: throw new IOException("Ort Result File is not in a supported format.");
+        }
+
+        JsonNode yamlContent = mapper.readTree(ortResultFile);
```

Comment:
Name yamlContent is not the right one anymore?

Reply:
Well, strictly speaking [YAML is a super-set of JSON](https://en.wikipedia.org/wiki/YAML) :stuck_out_tongue: But fixed still for good narure :wink: Thanks!